tests/widgetresize.js

```diff
@@ -0,0 +1,461 @@
+				expectedWidth: '120px'
+			} ) );
+
+			// --- top handlers ---
+
+			it( 'enlarges correctly with left-top handler', generateResizeTest( {
+				usedHandle: 'top-left',
+				movePointerBy: { x: -10, y: -10 },
+				expectedWidth: '120px'
+			} ) );
+
+			it( 'enlarges correctly with left-top handler, y axis only', generateResizeTest( {
+				usedHandle: 'top-left',
+				movePointerBy: { x: 0, y: -10 },
+				expectedWidth: '120px'
+			} ) );
+
+			it( 'enlarges correctly with right-top handler', generateResizeTest( {
+				usedHandle: 'top-right',
+				movePointerBy: { x: 10, y: -10 },
+				expectedWidth: '120px'
+			} ) );
+
+			it( 'enlarges correctly with right-top handler, y axis only', generateResizeTest( {
+				usedHandle: 'top-right',
+				movePointerBy: { x: 0, y: -10 },
+				expectedWidth: '120px'
+			} ) );
+		} );
+	} );
+
+	describe( 'integration (percents)', () => {
+		describe( 'side aligned widget', () => {
+			beforeEach( () => {
+				createResizer( {
+					unit: undefined
+				} );
+			} );
+
+			it( 'properly sets the state for subsequent resizes', () => {
+				const usedResizer = 'top-right';
+				const domParts = getWidgetDomParts( editor, widget, usedResizer );
+				const initialPointerPosition = getHandleCenterPoint( domParts.widget, usedResizer );
+
+				const intermediatePointerPosition = initialPointerPosition.clone().moveBy( 100, 0 );
+				resizerMouseSimulator.dragTo( editor, domParts.resizeHandle, intermediatePointerPosition );
+				sinon.assert.calledWithExactly( commitStub.firstCall, '50%' );
+
+				const finalPointerPosition = intermediatePointerPosition.clone().moveBy( 100, 0 );
+				resizerMouseSimulator.dragTo( editor, domParts.resizeHandle, finalPointerPosition );
+				sinon.assert.calledWithExactly( commitStub.secondCall, '75%' );
+
+				expect( commitStub.callCount ).to.be.equal( 2 );
+			} );
+
+			it( 'shrinks correctly with bottom-left handler', generateResizeTest( {
+				usedHandle: 'bottom-left',
+				movePointerBy: { x: 10, y: -10 },
+				expectedWidth: '22.5%'
+			} ) );
+		} );
+
+		describe( 'centered widget', () => {
+			beforeEach( () => {
+				createResizer( {
+					isCentered: () => true,
+					unit: undefined
+				} );
+			} );
+
+			it( 'shrinks correctly with bottom-left handler', generateResizeTest( {
+				usedHandle: 'bottom-left',
+				movePointerBy: { x: 10, y: -10 },
+				expectedWidth: '20%'
+			} ) );
+
+			it( 'enlarges correctly with bottom-right handler', generateResizeTest( {
+				usedHandle: 'bottom-right',
+				movePointerBy: { x: 0, y: 5 },
+				expectedWidth: '27.5%'
+			} ) );
+
+			it( 'enlarges correctly an image with unsupported width unit', () => {
+				editor.editing.view.change( writer => {
+					writer.setStyle( 'width', '100pt', widget );
+				} );
+
+				generateResizeTest( {
+					usedHandle: 'bottom-right',
+					movePointerBy: { x: 0, y: 5 },
+					expectedWidth: '36.67%'
+				} )();
+			} );
+		} );
+	} );
+
+	describe( '_getResizerByHandle', () => {
```

Comment:
```suggestion
	describe( '_getResizerByHandle()', () => {
```

Same for all other describes.

Reply:
Pushed a fix for that and few other occurrences.